modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraboot/TerraBootApiFetcher.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+package org.eclipse.xpanse.modules.deployment.deployers.terraform.terraboot;
+
+import jakarta.annotation.Resource;
+import java.util.List;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.deployment.deployers.terraform.terraboot.generated.api.RetrieveTerraformResultApi;
+import org.eclipse.xpanse.modules.deployment.deployers.terraform.terraboot.generated.model.ReFetchResult;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClientException;
+
+/** Bean to fetch api of terra-boot. */
+@Slf4j
+@Component
+public class TerraBootApiFetcher {
```

Comment:
Its better to call it -> TerraBootApiStoredResultsFetcher
Same for TofuMaker also